### ISSUE TITLE
Switched to SMS, can't switch back to signal

### ISSUE THREAD

**Issue by mitscherdinger** [signalapp/Signal-Android#9477]
Dear Developers!

I'm using signal on a Shift 5me smartphone without Google Apps. Installing Signal I decided to even deactivate the SMS/MMS-app from AOSP. Worked some time, but now I have this issue:
Contacting a friend with Signal with a brand new smartphone and a BackUp from Signal installed, I realized that the message was sent but not recieved. Waited a few days, nothing changed. I wanted to check if my friend had a problem with Signal and switched to SMS to ask what's going on. Some days later I got the message that his smartphone running Signal had to be repaired which took some days.
So, I tried to switch back to "save messaging" (=Signal-protocol) but the messages didn't even leave the phone. They also wouldn't if I send them in SMS-mode.
So, my smartphone is now in a stage I can't send any messages to this friend. I can send Signal-messages to other people but SMS via Signal seems completely dead. No sending, at least, even using Signal to send SMS to other contacts - don't know, if receiving still works…

Sending Signal-messages through the desktop version on the other hand still works fine.
For SMS I use the AOSP SMS/MMS-app, now, since I could not convince all my contacts using Signal. ;) Sending SMS with this other app works.

### Device info
**Device:** Shift 5me
**Android version:** 8.0.0
**Signal version:** 4.55.1

### Link to debug log
https://debuglogs.org/4a5d420d577a6f93c02f9732514a7cdef049b275af97828b6dce6d35497aaa38
It took some time since the first appearance of this issue. Thought, it could be a glitch that will be repaired by itself after some time… On March, the 5th I experienced the issue with the SMS that can't be sent.

**Comment by greyson-signal** [comment]
Messages within a thread are sent in order. It looks like you have an SMS message that isn't able to be sent, which is therefore blocking Signal messages later in that thread from being sent. It looks like it might not be getting sent because Signal doesn't think that you have cellular service.

As a temporary fix, try the following:
- In the affected thread, delete all "stuck" SMS messages (i.e. non-Signal messages)
- Go to Settings > "SMS and MMS" and enable Wi-Fi compatibility mode.

Hopefully your messages should get unstuck afterwards, maybe after restarting the app.

The real question is why the SMS got stuck to begin with (or more specifically, why Signal doesn't think you have cell service), but there's nothing helpful in the logs for that.

**Comment by Ollewa123** [comment]
Since OP never replied and I faced the exact same problem, I would like to report back that the temporary fix indeed solved the problem. I had to force close signal after the two steps described above but afterwards it worked fine again.

Additionally I want to add that, in my case at least, the problem wasn't isolated on one conversation. SMS messages weren't send in any conversation I tried, essentially blocking any outgoing messages (sms or native-signal) in these chats.

Thank you for the fix!
